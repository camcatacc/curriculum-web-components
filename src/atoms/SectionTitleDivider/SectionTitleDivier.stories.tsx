// Modules
import React from "react";

// Storybook Modules
import { ComponentStory, ComponentMeta } from "@storybook/react";

// Elements
import SectionTitleDivider from "atoms/SectionTitleDivider/SectionTitleDivider";

// Template
const Template: ComponentStory<typeof SectionTitleDivider> = (args) => <SectionTitleDivider {...args} />;

// Variants
export const Standard = Template.bind({});
Standard.args = {};

// Export
export default {
	title: "Atoms/SectionTitleDivider",
	component: SectionTitleDivider
} as ComponentMeta<typeof SectionTitleDivider>;
