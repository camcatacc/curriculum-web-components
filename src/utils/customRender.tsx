import { ThemeProvider } from "@material-ui/styles";
import { render, RenderOptions, RenderResult } from "@testing-library/react";
import { ReactElement } from "react";
import { getMaterialTheme } from "styles/materialUi";

export const customRender = (ui: ReactElement, renderOptions?: RenderOptions & { darkMode?: boolean }): RenderResult => {
	const Wrapper: React.FC = ({ children }) => {
		return <ThemeProvider theme={getMaterialTheme(renderOptions?.darkMode ? "dark" : "light")}>{children}</ThemeProvider>;
	};

	return render(ui, { wrapper: Wrapper, ...renderOptions });
};
