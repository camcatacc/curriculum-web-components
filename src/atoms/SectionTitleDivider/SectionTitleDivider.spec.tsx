// Modules
import React from "react";
import { customRender } from "utils/customRender";

// Elements
import SectionTitleDivider from "atoms/SectionTitleDivider/SectionTitleDivider";
import { ThemeProvider } from "@material-ui/styles";
import { getMaterialTheme } from "styles/materialUi";

// Tests
describe("SectionTitleDivider", () => {
	it("Renders as expected", () => {
		const { container } = customRender(<SectionTitleDivider />);
		expect(container).toMatchSnapshot();
	});

	it("Renders as expected with animation", () => {
		const { container } = customRender(<SectionTitleDivider animated={true} />);
		expect(container).toMatchSnapshot();
	});

	it("Renders white if dark mode is activated", () => {
		const { container } = customRender(
			<ThemeProvider theme={getMaterialTheme("dark")}>
				<SectionTitleDivider animated={true} />
			</ThemeProvider>
		);
		expect(container).toMatchSnapshot();
	});
});
