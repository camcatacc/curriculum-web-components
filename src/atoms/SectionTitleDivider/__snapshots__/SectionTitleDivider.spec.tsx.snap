// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionTitleDivider Renders as expected 1`] = `
<div>
  <div
    class="self-center border-b-2 border-black border-solid w-20 mt-4 mb-4"
  />
</div>
`;

exports[`SectionTitleDivider Renders as expected with animation 1`] = `
<div>
  <div
    class="self-center border-b-2 border-black border-solid w-20 mt-4 mb-4"
    style="opacity: 0; transform: translateX(-50px) translateY(0px) translateZ(0);"
  />
</div>
`;

exports[`SectionTitleDivider Renders white if dark mode is activated 1`] = `
<div>
  <div
    class="self-center border-b-2 border-white border-solid w-20 mt-4 mb-4"
    style="opacity: 0; transform: translateX(-50px) translateY(0px) translateZ(0);"
  />
</div>
`;
